### PR TITLE
Introduce e2e tests

### PR DESCRIPTION
**Pull request by RobertHebel**
Deployed to https://beghp.github.io/gh-pages-rc-12<br>* Removed not working gemini tests and config<br>* Introduce cypress tests<br>* Run e2e tests in the CI<br>* Create Docker and docker-compose files for dockerized e2e tests<br>* Update .eslint config so it works with cypress<br>* Update README so it includes actual information how to run unit and e2e tests<br><br>- [ ] `package.json:version` has been updated with `npm version patch` (or `major/minor` as appropriate)<br>- [ ] `@brainhubeu/react-carousel` version has been updated in `docs-www/package.json` to the same which is in `package.json:version`<br><br>

### CODE REVIEW

File: test/cypress/integration/Thumbnails.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+describe('Thumbnails', () => {
+  beforeEach(() => {
+    cy.visit('/docs/examples/thumbnails');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'mona');
+  });
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/Thumbnails.test.js]:
```suggestion
  });

```

File: test/cypress/integration/SwipeSlides.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      .trigger('mousemove', { pageX: 300 })
+      .trigger('mouseup', { force: true });
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'scream');
+  });
+
+  it(`Stays at the same slide if the previous one doesn't exist`, () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/SwipeSlides.test.js]:
```suggestion
  it(`stays at the same slide if the previous one doesn't exist`, () => {
```

File: test/cypress/integration/SwipeSlides.test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  beforeEach(() => {
+    cy.visit('/docs/examples/simpleUsage/');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'mona');
+  });
+
+  it('Swipe slide to the right if the next slide exist', () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/SwipeSlides.test.js]:
```suggestion
  it('swipes slide to the right if the next slide exist', () => {
```

File: test/cypress/integration/Dots.test.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+describe('Dots', () => {
+  beforeEach(() => {
+    cy.visit('/docs/examples/dots');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'mona');
+  });
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/Dots.test.js]:
```suggestion
  });

```

File: test/cypress/integration/ControlledInput.test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  it('Stays at the last slide if the value in input is bigger than number of slides', () => {
+    cy.get('input').type('50');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'starry-night');
+  });
+
+  it('Stays at the first slide if the value in input is less than 0', () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/ControlledInput.test.js]:
```suggestion
  it('stays at the first slide if the value in input is less than 0', () => {
```

File: test/cypress/integration/ControlledInput.test.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  it('Goes to the slide index provided in input', () => {
+    cy.get('input').type('2');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'starry-night');
+  });
+
+  it('Stays at the last slide if the value in input is bigger than number of slides', () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/ControlledInput.test.js]:
```suggestion
  it('stays at the last slide if the value in input is greater than number of slides', () => {
```

File: test/cypress/integration/ControlledInput.test.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+describe('Controlled input', () => {
+  beforeEach(() => {
+    cy.visit('/docs/examples/controlled/');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'mona');
+  });
+  it('Goes to the slide index provided in input', () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/ControlledInput.test.js]:
```suggestion

  it('Goes to the slide index provided in input', () => {
```

File: test/cypress/integration/ClickToChange.test.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      .children('img')
+      .click('left');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'scream');
+  });
+
+  it('allow to change slide when clicks on the previous slide', () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/ClickToChange.test.js]:
```suggestion
  it('allows to change slide when clicks on the previous slide', () => {
```

File: test/cypress/integration/ClickToChange.test.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
+describe('Click to change', () => {
+  beforeEach(() => {
+    cy.visit('/docs/examples/clickToChange/');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'mona');
+  });
+  it('allow to change slide when clicks on the next slide', () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/ClickToChange.test.js]:
```suggestion

  it('allows to change slide when clicks on the next slide', () => {
```

File: test/cypress/integration/Arrows.test.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      cy.get('.BrainhubCarousel__arrowRight')
+        .trigger('click');
+    }
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'starry-night');
+
+
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/Arrows.test.js]:
```suggestion
```

File: test/cypress/integration/Arrows.test.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    cy.get('.BrainhubCarousel__arrowLeft')
+      .trigger('click');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'mona');
+  });
+
+  it(`Does't allow to click on the arrow if there is no slides more`, () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/Arrows.test.js]:
```suggestion
  it(`does't allow to click on the arrow if there is no more slides`, () => {
```

File: .circleci/config.yml [hunk 12]
```diff
@@ -8,12 +8,23 @@ jobs:
...
-      - run: yarn test-unit
+      - run: yarn test:unit
       - run: yarn build
       - run:
           name: publish gh pages for a given PR
           command: ./tools/deploy-gh-pages-pr.sh
 
+  test-e2e:
+    docker:
+      - image: circleci/node:12.16.1
```
Comment (piotr-s-brainhub) [inline .circleci/config.yml]: couldn't you use `cypress/included:4.1.0` image here and run commands directly without Dockerfile and without docker-compose?

File: README.md [hunk 13]
```diff
@@ -391,20 +391,13 @@ render() {
...
-Install selenium if it is not installed
 ```
-npm run selenium-install
-```
-
-```
-1. npm run selenium-start
-2. npm run example
-3. npm run test-regression
+yarn test:e2e
```
Comment (piotr-s-brainhub) [inline README.md]: 👍
I like that one command can run e2e tests

File: .circleci/config.yml (hunk 12 above)
Comment (RobertHebel) [reply .circleci/config.yml]: I'm using docker cause I want to have the same test result independent on the OS/machine it's running in opposite to germini/hermione tests where we have got different results on the CI than on our local machines

File: test/cypress/integration/SwipeSlides.test.js [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  beforeEach(() => {
+    cy.visit('/docs/examples/simpleUsage/');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'mona');
+  });
+
+  it('swipe slide to the right if the next slide exist', () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/SwipeSlides.test.js]:
```suggestion
  it('swipes slide to the right if the next slide exists', () => {
```

File: test/cypress/integration/ControlledInput.test.js [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+  it('goes to the slide index provided in input', () => {
+    cy.get('input').type('2');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'starry-night');
+  });
+
+  it('stays at the last slide if the value in input is bigger than number of slides', () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/ControlledInput.test.js]:
```suggestion
  it('stays at the last slide if the value in input is greater than number of slides', () => {
```

File: test/cypress/integration/Arrows.test.js [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+    cy.get('.BrainhubCarousel__arrowLeft')
+      .trigger('click');
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'mona');
+  });
+
+  it(`does't allow to click on the arrow if there is no slides more`, () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/Arrows.test.js]:
```suggestion
  it(`does't allow to click on the arrow if there is no more slides`, () => {
```

File: test/cypress/integration/SwipeSlides.test.js [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+      .trigger('mousemove', { pageX: 300 })
+      .trigger('mouseup', { force: true });
+
+    cy.get('.BrainhubCarouselItem--active')
+      .children('img')
+      .should('have.attr', 'src')
+      .and('contain', 'scream');
+  });
+
+  it(`stays at the same slide if the previous one doesn't exists`, () => {
```
Comment (piotr-s-brainhub) [inline test/cypress/integration/SwipeSlides.test.js]:
```suggestion
  it(`stays at the same slide if the previous one doesn't exist`, () => {
```

**Pull request merged**